Xamarin.Forms.Xaml/XamlLoader.cs

```diff
@@ -152,7 +152,15 @@ static string GetXamlForType(Type type)
 
 			var typeInfo = type.GetTypeInfo();
 			var assembly = typeInfo.Assembly;
-			var resourceId = typeInfo.GetCustomAttribute<XamlResourceIdAttribute>()?.ResourceId;
+
+			string resourceId = null;
+			foreach (var xria in assembly.GetCustomAttributes<XamlResourceIdAttribute>()) {
```

Comment:
@jassmith please review this perf-wise, and tell me if you think it's useful to cache that info or not (I'm lending in the *not* direction)